spec/models/ems_event_spec.rb

```diff
@@ -519,4 +519,25 @@
       end
     end
   end
+
+  context 'Physical Storage Events' do
+    let(:ems)   { FactoryBot.create(:ems_autosde) }
+    let(:physical_storage) { FactoryBot.create(:physical_storage_autosde, :ems_ref => "ems1", :ext_management_system => ems) }
+    before do
+      @event_hash = {
+        :event_type               => "autosde_critical_alert",
+        :ems_ref                  => "1",
+        :physical_storage_ems_ref => physical_storage.ems_ref,
+        :ems_id                   => ems.id,
+        :message                  => "description"
+      }
+    end
```

Comment:
If you want to set up a shared variable it is preferable to use a `let(:event_hash) do; end` block over an ivar in a `before do; end` block.

```suggestion
    let(:event_hash) do
      {
        :event_type               => "physical_storage_alert",
        :ems_ref                  => "1",
        :physical_storage_ems_ref => physical_storage.ems_ref,
        :ems_id                   => ems.id,
        :message                  => "description"
      }
    end
```